proposals/XXXX-generic-subscripts.md

```diff
@@ -0,0 +1,71 @@
+# Generic Subscripts
+
+* Proposal: [SE-NNNN](NNNN-filename.md)
+* Authors: [Chris Eidhof](http://github.com/chriseidhof/)
+* Review Manager: TBD
+* Status: **Awaiting review**
+
+* Bugs: [SR-115](https://bugs.swift.org/browse/SR-115?jql=text%20~%20%22Generic%20subscript%22)
+
+## Introduction
+
+Make it possible to have generic subscripts. Example:
+
+```swift
+extension Collection {
+  subscript<Indices: Sequence>(indices: Indices) -> [Iterator.Element] where Indices.Iterator.Element == Index {
+    // ...
```

Comment:
I believe the closing braces are in the next hunk. Ditto for the other case below.